beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestIntegrationBase.java

```diff
@@ -1000,6 +1010,43 @@ public Void withHandle(final Handle handle) throws Exception {
         });
     }
 
+
+    protected boolean areAllNotificationsProcessed(final Long tenantRecordId) {
+        int nbNotifications = 0;
+        for (final NotificationQueue notificationQueue : notificationQueueService.getNotificationQueues()) {
+            final Iterator<NotificationEventWithMetadata<NotificationEvent>> iterator = notificationQueue.getFutureOrInProcessingNotificationForSearchKey2(null, tenantRecordId).iterator();
+            try {
+                while (iterator.hasNext()) {
+                    final NotificationEventWithMetadata<NotificationEvent> notificationEvent = iterator.next();
+                    if (!notificationEvent.getEffectiveDate().isAfter(clock.getUTCNow())) {
+                        nbNotifications += 1;
+                    }
+                }
+            } finally {
+                // Go through all results to close the connection
+                while (iterator.hasNext()) {
+                    iterator.next();
+                }
+            }
+        }
+        if (nbNotifications != 0) {
+            log.info(": {} queue(s) with more notification(s) to process", nbNotifications);
```

Comment:
Nit: the comment is incorrect, it's not the number of queues, but the number of notifications.